TMB unpaired samples (6/N)

AntoniaChroni opened the pull request:
<!--Hi there, thanks for your contribution! Please take a moment to fill out this template to facilitate the review of your pull request.-->

### Purpose/implementation Section

#### What scientific question is your analysis addressing?

Here, we explore TMB and generate summary statistics for unpaired longitudinal samples across cancer types. 

#### What was your approach?



#### What GitHub issue does your pull request address?

Closes https://github.com/d3b-center/pbta-tumor-evolution/issues/36

### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.

#### Which areas should receive a particularly close look?

Please review for script logic.


#### Is there anything that you want to discuss further?



#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

Yes.


### Results

#### What types of results are included (e.g., table, figure)?

The following plots are placed in the `plots_dir`: 
- `TMB-Bxp-stat-test.pdf`
- `TMB-unpaired-longitudinal-cg-id-sum.pdf`
- `TMB-unpaired-longitudinal.pdf`

The following tables are placed in the `results_dir`: 
- `stat_test.tsv`
- `stat_summary.tsv`

#### What is your summary of the results?

Overall, we have the highest number of samples for Diagnosis compared to the rest of the timepoints (1556). There is an indication of a TMB increase for Deceased specimens compared to Diagnosis for HGG samples. 


#### Reproducibility Checklist

<!-- Check all those that apply or remove this section if it is not applicable.-->

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.

#### Documentation Checklist

<!-- Please review https://github.com/d3b-center/pbta-tumor-evolution#documenting-your-analysis -->

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

<!-- IF YOUR PULL REQUEST IS A DATA RELEASE, PLEASE REMOVE THE [HTML COMMENT TAG](https://html.com/tags/comment-tag/) FROM THE SECTION BELOW AND COMPLETE THE CHECKLIST-->

<!-- 
### Data Release Checklist
- [ ] Is the table in doc/data-file-descriptions.md up to date?
- [ ] Is doc/data-format.md up to date?
- [ ] Is doc/release-notes.md up to date?
- [ ] Is download-data.sh up to date?
- [ ] Was download-data.sh tested and did it complete without error?
-->

Commit:
Create Rmd file for unpaired samples

Commit:
Clean code based on Lint

Commit:
Update bash file

Commit:
Minor edit about "remove hypermutant samples"

Commit:
Add ylim max(tmb)

Commit:
Update README file

Commit:
Add documentation for the Rmd

Commit:
Move short_histology_sum to load-process-inputs code chunk

Commit:
Add col_stat

Commit:
Add  summary-statistic code chunk and library(rstatix)

Commit:
Add multiple pairwise comparisons t-test

Commit:
Add summary stat and create bxp

Commit:
Remove test examples for calculate statistics

Commit:
Add stat.test

Commit:
Add stat.test results in the bxp

Commit:
Add link to the code of how to perform multiple paired t-tests

Commit:
Shorten code for bxp

Commit:
Clean code based on Lint

Commit:
Run locally Rmd and update plots

Commit:
Add tables with count per timepoint and cancer type

Commit:
Run Rmd to show #samples per timepoint and cancer type

Commit:
update html report

Commit:
Merge branch 'create-pptc-pdx-corplots' into achroni/tmb-unpaired-samples

Commit:
Calculate log10(tmb)

Commit:
Add code for ancer_group_display_n

Commit:
Plot log_10(tmb) and replace bxp with jitter

Commit:
Add code for creating jitter plot

Commit:
Add Medulloblastoma in the short_histology_sum

Commit:
Add code for Plot by using tumor_descriptor_cancer_n and run Rmd

Commit:
Run updated Rmd and upload html

Commit:
Remove plots from `plot-cancer-group-sum` code chunk and add part of this in the stat plots

Commit:
Edit Results section in README file

Commit:
Run Rmd in Docker

Commit:
Remove files generated by rstatix in folder v1

Commit:
Update script with keeping rstatix package

Commit:
Remove short_hist_sum, add tumor_descriptor_cgGFAC_n

Commit:
Save summary stats and remove cancer group plot with NA

Commit:
Save summary stat, remove extra ylim code

Commit:
Remove the link to the multiple paired t-tests

Commit:
Fix bug with not showing p.sign on the plot

Commit:
Run notebook in Docker

Commit:
Edit summary of cgGFAC and remove samples with less than 2

Commit:
Run in Docker

Commit:
Replace cancer_group with cg_id and edit ggplot code to reduce code lines

Commit:
Reduce code for ggplot

Commit:
Run new code in Docker

Commit:
Fix position of p value on the plot

Commit:
Run new code in Docker

Commit:
Remove not useful text

Commit:
Remove code chunk not useful

Commit:
Fix bug for df_plot_cgGFAC

Commit:
Run new code in Docker

Commit:
Remove not useful text

Commit:
Run new code in Docker

Commit:
Delete cancer_group_count.tsv

Commit:
Update README.md

Commit:
Edit authors

Commit:
Run in Docker

Commit:
Delete 05-variant-classification-plots.nb.html

Commit:
Remove code for cgGFAC

Commit:
Add cgGFAC in the select df

Commit:
Run new code in Docker

Commit:
Merge branch 'main' into achroni/tmb-unpaired-samples

Commit:
Merge branch 'main' into achroni/tmb-unpaired-samples

Commit:
Remove old folder v1

Commit:
Merge branch 'main' into achroni/tmb-unpaired-samples

Commit:
Merge branch 'main' into achroni/tmb-unpaired-samples

Commit:
Run in Docker after merge

Commit:
Merge branch 'main' into achroni/tmb-unpaired-samples

Commit:
Merge branch 'main' into achroni/tmb-unpaired-samples

Commit:
Merge branch 'main' into achroni/tmb-unpaired-samples

rjcorb commented on analyses/tmb-vaf-longitudinal/04-tmb-unpaired-samples-plot.Rmd:
```diff
@@ -0,0 +1,261 @@
+# Create df to use for plots
+df_plot <- df %>% 
+  left_join(timepoint_cg_n_df, by = c("tumor_descriptor", "cg_id")) %>%
+  filter(!tumor_descriptor_cg_n <= 2) %>% # remove if total #samples per timepoint is less than 2
```
I think this should be: 
```suggestion
  filter(!timepoint_cg_n <= 2) %>% # remove if total #samples per timepoint is less than 2
```

AntoniaChroni replied:
I edited the names between label and number to avoid confusion. See d4de754f89b9772726e258659abab95277f02874

rjcorb commented on analyses/tmb-vaf-longitudinal/04-tmb-unpaired-samples-plot.Rmd:
```diff
@@ -0,0 +1,261 @@
+         cg_multiple, cg_id, cgGFAC, cancer_group, short_histology, tmb, mutation_count, DNA) %>%
+  dplyr::mutate(td_cgGFAC = case_when(grepl("Deceased", tumor_descriptor) ~ "xDeceased", 
+                                      TRUE ~ tumor_descriptor),
+                log10_tmb = abs(log10(tmb))) %>% 
```
```suggestion
                log10_tmb = log10(tmb)) %>% 
```
I don't think the `abs()` should be included here...?

AntoniaChroni replied:
Most TMB values are less than one. When a number is a fraction (less than one), then the log is negative. To avoid negative values in the plots, I took the absolute of them. Does this make sense? @rjcorb

rjcorb commented on analyses/tmb-vaf-longitudinal/04-tmb-unpaired-samples-plot.Rmd:
```diff
@@ -0,0 +1,261 @@
+        geom_hline(yintercept = 0, size = 0.15) + 
+        scale_color_manual(values = palette, 
+                           breaks = sort(names(palette))) +
+        stat_summary(color = "black", size = 0.3) + 
```
Since there is no summary function supplied the default `mean_se()` is used, but it might be good to define the function in the code so it is noted.

AntoniaChroni replied:
Done, see 831e3a44e42a697c29bd3018afc841d7dae075ab

rjcorb commented on analyses/tmb-vaf-longitudinal/04-tmb-unpaired-samples-plot.Rmd:
```diff
@@ -0,0 +1,261 @@
+# Paired t-test is used because we have repeated measures by time
+stat.test <- df_plot_cgGFAC %>%
+  group_by(cgGFAC) %>%
+  pairwise_t_test(log10_tmb ~ tumor_descriptor_cgGFAC_n, 
+                  paired = FALSE, 
+                  p.adjust.method = "bonferroni") %>% 
```
you might consider doing a `wilcox_paired_test()` instead, to avoid parametric assumptions of t tests. Also, is `paired = FALSE` correct here?

AntoniaChroni replied:
That is a great point, thanks! This is now implemented. 
`paired = FALSE` was used here because the number of arguments per group (i.e., timepoint_cgGFAC_n) is uneven, e.g., `Atypical Teratoid Rhabdoid Tumor_Recurrence  (N=3)` vs `Atypical Teratoid Rhabdoid Tumor_Progressive  (N=10)`. 
`Paired = TRUE` could be used more in cases with before/after.

rjcorb commented on analyses/tmb-vaf-longitudinal/04-tmb-unpaired-samples-plot.Rmd:
```diff
@@ -0,0 +1,261 @@
+df_plot_cgGFAC <- df_plot %>% 
+  arrange(tumor_descriptor_cgGFAC_n) 
+
+df_plot_cgGFAC %>% 
+  select(cgGFAC, tumor_descriptor_cgGFAC_n, log10_tmb) %>% 
+  sample_n_by(cgGFAC, tumor_descriptor_cgGFAC_n, log10_tmb, size = 1) 
```
I think this is too long of a table to include in html

AntoniaChroni replied:
It is meant to be exploratory rather than a result worth saving in the repo.

rjcorb reviewed (commented):
These plots look great! `facet_wrap()` is a real game changer. I was a bit confused by the title "unpaired samples" because it made me think you were only going to be looking at samples that don't have a matched time point in the data set. Maybe something like "TMB by tumor histology and time point" would be better? I left a few other in-line comments.

naqvia reviewed (changes requested):
The code logic makes sense. I don't have anything more to add to what Ryan had already stated, but a minor thing would be that some of the plots have underscores (ie. in x and y axis labels and legends- `plots/TMB-Bxp-stat-test.pdf`), and need minor modifications.

AntoniaChroni commented:
> These plots look great! `facet_wrap()` is a real game changer. I was a bit confused by the title "unpaired samples" because it made me think you were only going to be looking at samples that don't have a matched time point in the data set. Maybe something like "TMB by tumor histology and time point" would be better? I left a few other in-line comments.

Thanks! Here, we explore the `unpaired longitudinal samples across cancer types`, as mentioned in the PR. But essentially these are all samples from the PBTA cohort. I edited the title in the notebook to reflect this. See a3cbab22d9f9a1542b050dd7b724435a3c1160db

Commit:
Edit title of notebook

Commit:
Edit timepoint_cg and timepoint_cgGFAC

Commit:
Replace labels with timepoint and edit labels for boxplots

AntoniaChroni commented:
> The code logic makes sense. I don't have anything more to add to what Ryan had already stated, but a minor thing would be that some of the plots have underscores (ie. in x and y axis labels and legends- `plots/TMB-Bxp-stat-test.pdf`), and need minor modifications.

Thanks for noticing, I edited the labels in the boxplot to match the rest. See cf9f99aac28ce6b662196a75605fcf1ac65b7c55

Commit:
Add the link to the test

Commit:
Remove the link to the wilcox test

Commit:
Add boxplot with considering effect size

Commit:
Add function rotate()

Commit:
Add fun.data = mean_se

Commit:
Remove td_cgGFAC and xDeceased

Commit:
Replace td_cgGFAC with tumor_descriptor

Commit:
Edit filename for boxplots

Commit:
Remove code for estimating effect size

Commit:
Add loop to run stats per cancer_group

Commit:
Remove , dodge = 1, step.increase = .01 for add_xy_position

Commit:
Edit indents in the loop

Commit:
Remove rotate()

Commit:
Save stat.summary and sta.test as tsv

Commit:
Update run-tmb-vaf-longitudinal.sh

Commit:
Fix typo with cancer_groups

Commit:
Replace timepoint_cgGFAC_n

Commit:
Replace df_plot_cgGFAC with df_plot

Commit:
Update 04-tmb-unpaired-samples-plot.Rmd

Commit:
Remove code for timepoint_cg_n_df

This is not used anymore in this script.

Commit:
Arrange df_sub and add timepoint_cgGFAC_n in the boxplot

Commit:
Add cgGFAC in the stat.test table

Commit:
Add rotate and remove axis.text.x, edit step.increase and change file size

Commit:
Remove step.increase, add max value for yaxis, edit figure size

Commit:
Run in Docker with new code

Commit:
chnage the order of filtering single timepoints and calculating timepoint_cgGFAC_n_df

Commit:
Run in Docker new code

AntoniaChroni commented:
I split the boxplots by cancer groups and edited some parts of the code to match that. I also fixed a bug that was miscalculating the number of counts for `timepoint_cgGFAC`. @rjcorb @naqvia

rjcorb commented on analyses/tmb-vaf-longitudinal/04-tmb-unpaired-samples-plot.Rmd:
```diff
@@ -0,0 +1,292 @@
+  select(Kids_First_Participant_ID, Kids_First_Biospecimen_ID, tumor_descriptor,
+         broad_histology, match_id_DNA, experimental_strategy, 
+         cg_multiple, cg_id, cgGFAC, cancer_group, short_histology, tmb, mutation_count, DNA) %>%
+  dplyr::mutate(log10_tmb = abs(log10(tmb))) %>% 
```
I still think the `abs()` needs to be removed here.

AntoniaChroni replied:
Done, see d469cc1fee5c7c2d5c0c378f70ee96c44fb19a28

rjcorb commented on analyses/tmb-vaf-longitudinal/04-tmb-unpaired-samples-plot.Rmd:
```diff
@@ -0,0 +1,292 @@
+    add_xy_position(x = "timepoint_cgGFAC_n") # , step.increase = .08
+
+  # Create bxp
+  print(ggboxplot(df_sub,
+                  x = "timepoint_cgGFAC_n",
+                  y = "log10_tmb",
+                  color = "tumor_descriptor",
+                  palette = palette,
+                  add = c("mean")) + # add = c("mean", "jitter")
+          theme_Publication() + 
+          #rotate() +
+          theme(axis.text.x = element_text(angle = 90)) +
+          stat_pvalue_manual(stat.test,
+                             #tip.length = 0.01,
+                             hide.ns = TRUE, 
+                             label = "p.adj.signif",
+                             y.position = "y.position") +
+          scale_y_continuous(limits = c(0, max(stat.test$y.position))) +
+          labs(title = cancer_groups[i],
+               x = "Tumor descriptor",
+               y = "log10TMB",
+               color = "Timepoint"))
```
I think there is an issue with the significance values in the plot not matching those in the results table. If you look at the plot for Other tumor there are significant TMB differences reported between Progressive and other time point, but the stats indicate these are actually differences between Diagnosis and other timepoints: 

```
         .y.              group1                   group2  n1  n2 statistic        p    p.adj p.adj.signif cgGFAC y.position       groups xmin xmax
1  log10_tmb     Deceased (N=25)        Diagnosis (N=920)  25 920    8887.0 5.20e-02 2.60e-01           ns  Other    1.63988 Deceased....    1    2
2  log10_tmb     Deceased (N=25)      Progressive (N=122)  25 122    1257.5 1.68e-01 5.04e-01           ns  Other    1.73308 Deceased....    1    3
3  log10_tmb     Deceased (N=25)       Recurrence (N=113)  25 113    1356.5 7.59e-01 8.18e-01           ns  Other    1.82628 Deceased....    1    4
4  log10_tmb     Deceased (N=25) Second Malignancy (N=13)  25  13     221.0 7.40e-02 2.97e-01           ns  Other    1.91948 Deceased....    1    5
5  log10_tmb   Diagnosis (N=920)      Progressive (N=122) 920 122   58699.0 4.09e-01 8.18e-01           ns  Other    2.01268 Diagnosi....    2    3
6  log10_tmb   Diagnosis (N=920)       Recurrence (N=113) 920 113   65360.0 7.67e-06 7.67e-05         ****  Other    2.10588 Diagnosi....    2    4
7  log10_tmb   Diagnosis (N=920) Second Malignancy (N=13) 920  13    9349.5 4.73e-04 4.00e-03           **  Other    2.19908 Diagnosi....    2    5
8  log10_tmb Progressive (N=122)       Recurrence (N=113) 122 113    8221.0 1.10e-02 7.50e-02           ns  Other    2.29228 Progress....    3    4
9  log10_tmb Progressive (N=122) Second Malignancy (N=13) 122  13    1200.5 2.00e-03 1.90e-02            *  Other    2.38548 Progress....    3    5
10 log10_tmb  Recurrence (N=113) Second Malignancy (N=13) 113  13    1031.0 1.80e-02 1.06e-01           ns  Other    2.47868 Recurren....    4    5
```
I am not 100% sure, but this may be caused by different leveling of factors in `stat.test` vs. `df_sub`. in df_sub (and plot) it is Diagnosis, Progressive, Recurrence, Deceased, Second Malignancy, but in stat.test it is alphabetical (so Deceased first).

AntoniaChroni replied:
Thanks for catching this! `df_sub` was arranged by `tumor_descriptor` and `timepoint_cgGFAC_n`. See following code:

```
df_sub <- df_plot %>% 
    filter(cgGFAC == cancer_groups[i]) %>% 
    arrange(tumor_descriptor, timepoint_cgGFAC_n) 
```


I adjusted `stat.test` to match this but it didn't work (see 616fec8e2ea7443d65148226f7f0047d9257a41e). It turned out to be a bug related to the `add_xy_position()` function that misplaces the brackets on the ggplot (reported here https://github.com/kassambara/ggpubr/issues/477). I replaced it with `add_y_position()` (see 35e39dba3d131c89d3513e7e7abcf22c23f8d5e0) and the problem was fixed. :)  @rjcorb

rjcorb reviewed (changes requested):
Most of the code seems to be in good shape, but see comment about stats not matching up between table and plot. I also think the abs() needs to be removed when calculating log10 TMB

naqvia reviewed (approved):
Plots look much better. In addition to Ryans suggestion, and something minor that could be added is `set_seed() ` call anytime you use `geom_jitter()`  when plotting. It will ensure that the dots are always in the same position, right now they slightly change positions.

Commit:
Replace abs with -log10

AntoniaChroni commented:
> bout stats not matching up between table and plot. I also think the abs() needs to be removed when calculating log10 TMB

I have replace this with `-log10`, see 27106b8c338e5bea9f2850285f20db5997d21711

Commit:
Add set.seed for geom_jitter()

AntoniaChroni commented:
> geom_jitter()

@naqvia Thanks for noticing! This has been now addressed here 1e1534371919184b5feb56bd7a506f1feda1875f

Commit:
Arrange order of timepoints

Commit:
Move Add add_xy_position code earlier so to be saved in the tsv files

Commit:
Fix bug with wrong position of brackets

Commit:
Run new code in Docker

Commit:
Remove negative from log10(tmb)

Commit:
Set min, max values for y axis for log10_tmb

Commit:
Add log10_ylim_min in boxplot

Commit:
Add rotate and edit figure size

Commit:
Add 0.5 to ylim_max

Commit:
Remove rotate, adjust figure boxplot size, remove 0.5 from ylim

Commit:
Run new code in Docker

Commit:
Add code to save Wilcox-test

Commit:
Remove label = "p.adj.signif" code

Commit:
Run in Docker

rjcorb reviewed (approved):
This looks great! My only suggestion would be to make the points larger in the histology-specific TMB plots. But I am approving and don't need to review again.

Commit:
Add str_wrap to edit name of `cg_id` to fit in the plot title for the panel

Commit:
Run new code in Docker

AntoniaChroni commented:
> This looks great! My only suggestion would be to make the points larger in the histology-specific TMB plots. But I am approving and don't need to review again.

@rjcorb  @jharenza This has been now addressed here: ea83984d71f4fb5534e9136a232279ae56bb40cf

jharenza reviewed (approved):
Hi @AntoniaChroni thanks for all of your hard work on this! It looks great. My only suggestion is to try to make sure all of the x-axes and legends are in the same order (for example for some of the sina plots, I am seeing deceased coming before another phase of therapy, when I might expect them to be listed last, and they are in the legend usually. Otherwise, I don't need to see this again! Nice work!

Commit:
Re-order legend and replace name in the x axis

Commit:
Replace name for df to use for plot and code for boxplot

Commit:
Add if else statement to order DMG, HGG

Commit:
Run new code in Docker

Commit:
Order ATRT boxplots

Commit:
Run new code with ordering ATRT

AntoniaChroni commented:
> Hi @AntoniaChroni thanks for all of your hard work on this! It looks great. My only suggestion is to try to make sure all of the x-axes and legends are in the same order (for example for some of the sina plots, I am seeing deceased coming before another phase of therapy, when I might expect them to be listed last, and they are in the legend usually. Otherwise, I don't need to see this again! Nice work!

@jharenza Thanks! These are now addressed here: 58622501ca1dfadbe5bfec1e92c5a161932cffe9, 89eccc111f36791bb7554938f777ba818d92e50e and 105ae3932a116ac742e79051391ac4500ed5e587
